src/components/DownloadButton/DownloadButton.tsx

```diff
@@ -51,49 +88,51 @@ const DownloadButton = React.memo((props: DownloadButtonProps) => {
     }
   }, [userAgentData, os])
 
-  const defaultLinks = identityId
-    ? getCDNRelease(CDNSource.AUTO_SIGNING, identityId)
-    : getCDNRelease(CDNSource.LAUNCHER)
-  const links = cdnLinks || defaultLinks
+  // Use LAUNCHER by default (without identityId) until user clicks
+  const links = useMemo(() => {
+    return cdnLinks || getCDNRelease(CDNSource.LAUNCHER)
+  }, [cdnLinks])
 
   const defaultDownloadOption: DownloadOption | null = useMemo(() => {
-    if (
-      !userAgentData ||
-      !links ||
-      !links[userAgentData.os.name] ||
-      !links[userAgentData.os.name][userAgentData.cpu.architecture]
-    ) {
-      return null
-    }
-
-    if (
-      userAgentData.os.name === OperativeSystem.MACOS &&
-      links[userAgentData.os.name][userAgentData.cpu.architecture]
-    ) {
-      return {
-        icon: <DownloadButtonAppleIcon />,
-        link: links[userAgentData.os.name][userAgentData.cpu.architecture],
-        arch: userAgentData.cpu.architecture,
-      }
-    }
-
-    return {
-      icon: <DownloadButtonWindowsIcon />,
-      link: links[userAgentData.os.name][userAgentData.cpu.architecture],
-      arch: userAgentData.cpu.architecture,
-    }
+    return createDownloadOption(links, userAgentData)
   }, [userAgentData, links])
 
   const finalHref = useMemo(() => {
     return href || defaultDownloadOption?.link || config.get("DOWNLOAD_URL")
   }, [href, defaultDownloadOption])
 
   const handleClick = useCallback(
-    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+    async (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
       event.preventDefault()
 
+      setIsDownloading(true)
+
+      // Always try to get identityId on-demand when clicking
+      let currentIdentityId: string | undefined
+      if (getIdentityId) {
+        try {
+          currentIdentityId = await getIdentityId()
+        } catch (error) {
+          console.error("Failed to generate identityId:", error)
+          // Continue with fallback behavior
+        }
+      }
+
+      // Recalculate links with the identityId (if available) or use LAUNCHER as fallback
+      const currentLinks =
+        cdnLinks ||
```

Comment:
Shouldn't we prefer the links with the `currentIdentityId` instead of the `cdnLinks`  when the `currentIdentityId` is set?